.github/workflows/test.yml

```diff
@@ -0,0 +1,27 @@
+name: "Tests"
+
+on: [pull_request]
+
+
+jobs: 
+    test:
+        name: "Unit & E2E Test"
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                php-versions: ['7.4', '8.0','nightly']
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+              with:
+                fetch-depth: 2  
+            - name: Setup PHP Action
+              uses: shivammathur/setup-php@2.25.5
```

Comment:
Let's use docker following the workflow [here](https://github.com/utopia-php/database/blob/main/.github/workflows/tests.yml) so we don't need to use third-party actions